core-libs/setup/ssr/optimized-engine/rendering-strategy-resolver.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Request } from 'express';
+import { RenderingStrategy } from './ssr-optimization-options';
+import { RenderingStrategyResolverOptions } from './rendering-strategy-resolver-options';
+
+const hasExcludedParams = (
+  request: Request,
+  excludedParams: string[] | undefined
+): boolean => {
+  const params: string[] = request.query
+    ? Object.getOwnPropertyNames(request.query)
+    : [];
+
+  if (!excludedParams) {
+    return false;
+  }
+
+  return excludedParams.some((excludedParam: string) =>
+    params.some((param: string): boolean => excludedParam === param)
+  );
+};
+
+const hasExcludedUrl = (
+  request: Request,
+  excludedUrls: string[] | undefined
+) => {
+  return request.url && excludedUrls
+    ? excludedUrls.some((url) => request.url.search(url) > -1)
+    : false;
+};
+
+const shouldFallbackToCsr = (
+  request: Request,
+  { excludedParams, excludedUrls }: RenderingStrategyResolverOptions
+) => {
+  return (
+    hasExcludedParams(request, excludedParams) ||
+    hasExcludedUrl(request, excludedUrls)
+  );
+};
+
+export const defaultRenderingStrategyResolver =
```

Comment:
`could:` consider to add the jsdoc

Reply:
The description of the `return` doesn't look good in the doc popup:
![image](https://github.com/SAP/spartacus/assets/28891782/670bb46b-e932-42fb-a160-9701ed172408)

Apart from that, I wondering whether it's necessary to add `excludedUrls` and `excludedParams` to the description (or at least shouldn't be marked `@param` as they are not params directly). The same thought about `request` - it's not a direct param of the resolver.

Reply:
as discussed I've left options params but removed the request